### ISSUE TITLE
Bug: [member-ordering] optionalityOrder is backwards

### ISSUE THREAD

**Issue by RedGuy12** [typescript-eslint/typescript-eslint#6267]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.4&sourceType=module&code=JYOwLgpgTgZghgYwgAgGIHt3IN4FgBQyyAJgPwBcyAzmFKAOYDcBRclNdITLyARu7QbNCyBAM7d8AXyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AW0XYCNFpaA9tAAmfZgHN0YANrhsEAEwAaOfKzzskUQDMAhgnxT1GzQJqUBTXYyIB5EXymQz+C1fi1tlaMkOqNAL7+EEHyALpyQQFAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
interface Foo {
  optional?: string;
  required: string;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/member-ordering": [
      2,
      {
        "default": {
          "optionalityOrder": "optional-first"
        }
      }
    ]
  },
};
```

### Expected Result

I expected no errors since I have `optionalityOrder` set to `optional-first`.

### Actual Result

There was an error on line 2 - `Member optional should be declared after all required members. 2:3 - 2:21`.

### Additional Info

This is also broken vice-versa: when `optionalityOrder` is set to `required-first` it requires all optional members first.

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/6255
